modules/ggl-sdk/src/object.c

```diff
@@ -3,105 +3,171 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include <assert.h>
+#include <ggl/attr.h>
 #include <ggl/buffer.h>
+#include <ggl/log.h>
 #include <ggl/object.h>
 #include <string.h>
-#include <stdalign.h>
 #include <stdbool.h>
 #include <stdint.h>
 
-typedef struct {
-    union {
-        bool boolean;
-        int64_t i64;
-        double f64;
-        GglBuffer buf;
-        GglList list;
-        GglMap map;
-    };
-
-    uint8_t type;
-} GglObjectPriv;
-
 static_assert(
-    sizeof(GglObject) == sizeof(GglObjectPriv), "GglObject impl invalid."
+    sizeof(void *) == sizeof(size_t),
+    "Platforms where pointers and size_t are not same width are not currently "
+    "supported."
 );
+
 static_assert(
-    alignof(GglObject) == alignof(GglObjectPriv), "GglObject impl invalid."
+    (sizeof(size_t) == 4) || (sizeof(size_t) == 8),
+    "Platform not 32 or 64 bits?"
```

Comment:
nit: I think this message should be more clear, e.g. Say only 32 or 64 bit systems are currently supported, if that's what we are verifying

Reply:
Its not entirely sure, since we're assuming pointers are word sized. This will also trip on CHERI, which we can add support for at some point if we have a customer.

Reply:
I see. Either way is fine then